app/components/edit/EditServiceChildCollection.tsx

```diff
@@ -0,0 +1,120 @@
+import React, {
+  FC, useState, useRef, useEffect,
+} from 'react';
+
+/**
+ * Renders a child property collection of a Service as a list of individual components,
+ * called ResourceObjectItem. Enables the user to add, modify, and delete members of
+ * the collection and save to the DB.
+ *
+ * In the future, this component could possibly be abstracted to replace the
+ * EditCollection HOC class-based JSX component that does essentially the same thing
+ * for child property collections of Resources.
+ */
+interface ComponentProps {
+  index: number;
+  item: any;
+  handleItemChange: any;
+}
+
+export const EditServiceChildCollection = ({
+  initialCollectionData,
+  handleCollectionChange,
+  ResourceObjectItem,
+  label,
+  blankTemplateObj,
+  buttonText,
+  propertyKeyName,
+}: {
+  initialCollectionData: any[];
+  handleCollectionChange: (field: any, value: any) => null;
```

Comment:
This would be a perfect application of generics. If you define the component with a type parameter, say `T`, then you can guarantee that both this collection type (`T[]`) and the handle change function (`(field: string, value: T) => void`) work on objects of that type as well.

Note that there's a bit of an issue with defining generic arrow functions in a TSX file because of a conflict between the TypeScript and JSX syntaxes, so you may have to apply [a workaround](https://stackoverflow.com/a/45576880) to actually type it out.

Reply:
I added a generic here. The only part where I'm a little unsure is using the `ResourceItem` interface defined on line 21 to extend `T` generic. I needed this as the `createItemComponents` method assumes that all members (T) of the collection will contain either an `id` or `keyId` prop. The `keyId` prop is actually added in the `addItem` method. It's needed for new items created on the client that don't have IDs assigned to them by the DB. Anyhow, let me know if you see any red flags with my implementation here. Thank you!